sdk/identity/azure-identity-broker/azure/identity/broker/_user_password.py

```diff
@@ -0,0 +1,85 @@
+        will cache tokens in memory.
+    :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
+    :keyword bool disable_instance_discovery: Determines whether or not instance discovery is performed when attempting
+        to authenticate. Setting this to true will completely disable both instance discovery and authority validation.
+        This functionality is intended for use in scenarios where the metadata endpoint cannot be reached, such as in
+        private clouds or Azure Stack. The process of instance discovery entails retrieving authority metadata from
+        https://login.microsoft.com/ to validate the authority. By setting this to **True**, the validation of the
+        authority is disabled. As a result, it is crucial to ensure that the configured authority host is valid and
+        trustworthy.
+    :keyword bool allow_broker: An authentication broker is an application that runs on a user's machine that manages
+        the authentication handshakes and token maintenance for connected accounts. The Windows operating system uses
+        the Web Account Manager (WAM) as its authentication broker. If this parameter is set to True, the broker will
+        be used when possible. Defaults to False.
+        Check https://learn.microsoft.com/azure/active-directory/develop/scenario-desktop-acquire-token-wam for more
+        information about WAM.
+    :keyword List[str] additionally_allowed_tenants: Specifies tenants in addition to the specified "tenant_id"
+        for which the credential may acquire tokens. Add the wildcard value "*" to allow the credential to
+        acquire tokens for any tenant the application can access.
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/credential_creation_code_snippets.py
+            :start-after: [START create_username_password_credential]
+            :end-before: [END create_username_password_credential]
+            :language: python
+            :dedent: 4
+            :caption: Create a UsernamePasswordCredential.
+    """
+
+    def __init__(self, client_id: str, username: str, password: str, **kwargs: Any) -> None:
+        # The base class will accept an AuthenticationRecord, allowing this credential to authenticate silently the
+        # first time it's asked for a token. However, we want to ensure this first authentication is not silent, to
+        # validate the given password. This class therefore doesn't document the authentication_record argument, and we
+        # discard it here.
+        self._allow_broker = kwargs.pop("allow_broker", None)
+        kwargs.pop("authentication_record", None)
+        super(UsernamePasswordBrokerCredential, self).__init__(
+            client_id=client_id, username=username, password=password, **kwargs
+        )
+        self._username = username
+        self._password = password
+
+    @wrap_exceptions
+    def _request_token(self, *scopes: str, **kwargs: Any) -> Dict:
```

Comment:
Seems like we only need to override _get_app and not _request_token in this case, since msa_passthrough and window handles aren't relevant for this credential and nothing change here.  Pretty much just need the same _get_app we have in the browser credential.